bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/PerspectiveExtensionReader.java

```diff
@@ -103,6 +106,29 @@ private boolean processActionSet(IConfigurationElement element) {
 	 * Process an extension. Assumption: Extension is for current perspective.
 	 */
 	private boolean processExtension(IConfigurationElement element) {
+		String[] attributes = element.getAttributeNames();
+		for (String attribute : attributes) {
+			boolean result = false;
+			switch (attribute) {
+			case IWorkbenchRegistryConstants.ATT_EDITOR_ONBOARDING_TEXT:
+				result = processEditorOnboardingText(element.getAttribute(attribute));
+				break;
+			case IWorkbenchRegistryConstants.ATT_EDITOR_ONBOARDING_IMAGE:
+				result = processEditorOnboardingImage(element, element.getAttribute(attribute));
+				break;
+			case IWorkbenchRegistryConstants.ATT_TARGET_ID:
+				result = true;
+				break;
+
+			default:
+				break;
+			}
+			if (!result) {
+				WorkbenchPlugin.log("Unable to process attribute: " + //$NON-NLS-1$
```

Comment:
Minor comment: For new code I think it better to use ILog.error()

Reply:
changed it.